### PR TITLE
Add Bugsnag to report errors

### PR DESCRIPTION
**Pull request by felipecruz91**

### CODE REVIEW

**Comment by rumpl** [comment]
QQ: can't all of the `bugnsnag.Notify` be done in one place? Like in a middleware or something?

**Review by benja-M-1 — COMMENTED** [review]
LGTM 👍 

I managed to trigger an error by removing the container that was used to clone a volume.

I left a small comment on the code.

I was expecting to have it configured in the frontend as well. I think it is valuable to have it even if the errors overlap the backend part:
- they can provide more context when it comes to debug (for example, what button has been clicked)
- also we might be surprised by some errors poping in, from my experience, it is better to have error monitoring with no errors rather than errors not monitored, especially when users can click on buttons and fill inputs 😅 

Also, as an improvement, we could add more versions in the data to monitor if a bug is correctly fixed across versions 
- the Docker Desktop version
- the Docker Extension backend version
- the versions of the @docker packages (when we have bugsnag in the frontend)                                

Finally, should we [identify users](https://docs.bugsnag.com/platforms/go/net-http/#identifying-users) the same way we do it for events? It could be interesting when debugging to see that prior to one error the user had another one for example.

**Comment by felipecruz91** [comment]
> Also, as an improvement, we could add more versions in the data to monitor if a bug is correctly fixed across versions
> - the Docker Desktop version
> - the Docker Extension backend version
> - the versions of the https://github.com/docker packages (when we have bugsnag in the frontend)

- Added the Docker Desktop version.

image

- I can't include the Extension backend version as I'd need to ship the Extensions CLI and the repo is private.

File: vm/main.go [hunk 1]
```diff
@@ -25,35 +27,58 @@ func main() {
...
+					_ = bugsnag.Notify(err, c.Request().Context())
+				}
+			} else {
+				// log to container logs
+				log.Error(err)
+				// log to Bugsnag
+				_ = bugsnag.Notify(err, c.Request().Context())
+			}
+		}
+		router.DefaultHTTPErrorHandler(err, c)
```
[the remark this answers is not in the document]
Comment (felipecruz91) [reply vm/main.go]: Moved to the `setup` pkg.

File: vm/internal/setup/bugsnag.go [hunk 2]
```diff
@@ -44,3 +47,32 @@ func ConfigureBugsnagHandler(server *http.Server) {
...
+				// log to Bugsnag
+				_ = bugsnag.Notify(err, c.Request().Context())
+			}
+		} else {
+			// log to container logs
+			log.Error(err)
+			// log to Bugsnag
+			_ = bugsnag.Notify(err, c.Request().Context())
+		}
+	}
```
Comment (benja-M-1) [inline vm/internal/setup/bugsnag.go:66]: nit: with early returns, I found the code more readable
- there are 2 conditional paths clearly identified 
- no need to look for the if when reading the code in the else block
- the happy path is aligned on the left, not surrounded by brackets on the 3rd level of indentation

I found this article interesting https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88

```suggestion
	if os.Getenv("BUGSNAG_API_KEY") == "" {
		return
	}
	
	he, ok := err.(*echo.HTTPError); 
	if ok && he.Code != http.StatusInternalServerError {
		return
	}
	
	// log to container logs
	log.Error(err)
	// log to Bugsnag
	_ = bugsnag.Notify(err, c.Request().Context())
```

**Review by benja-M-1 — APPROVED** [review]
One nit but other than it lgtm 👍 Can't wait to have this deployed

File: vm/internal/setup/bugsnag.go (hunk 2 above)
Comment (felipecruz91) [reply vm/internal/setup/bugsnag.go]: It's definitely more readable, thanks!

**Pull request merged**